extensions/sftp/processors/ListSFTP.cpp

```diff
@@ -261,20 +263,19 @@ void ListSFTP::onSchedule(const std::shared_ptr<core::ProcessContext> &context,
   }
   context->getProperty(TargetSystemTimestampPrecision.getName(), target_system_timestamp_precision_);
   context->getProperty(EntityTrackingInitialListingTarget.getName(), entity_tracking_initial_listing_target_);
-  if (!context->getProperty(MinimumFileAge.getName(), value)) {
-    logger_->log_error("Minimum File Age attribute is missing or invalid");
+
+  if (auto minimum_file_age = context->getProperty<core::TimePeriodValue>(MinimumFileAge)) {
+    minimum_file_age_ = minimum_file_age->getMilliseconds();
   } else {
-    core::TimeUnit unit;
-    if (!core::Property::StringToTime(value, minimum_file_age_, unit) || !core::Property::ConvertTimeUnitToMS(minimum_file_age_, unit, minimum_file_age_)) {
-      logger_->log_error("Minimum File Age attribute is invalid");
-    }
+    logger_->log_error("Minimum File Age attribute is missing or invalid");
   }
-  if (context->getProperty(MaximumFileAge.getName(), value)) {
-    core::TimeUnit unit;
-    if (!core::Property::StringToTime(value, maximum_file_age_, unit) || !core::Property::ConvertTimeUnitToMS(maximum_file_age_, unit, maximum_file_age_)) {
-      logger_->log_error("Maximum File Age attribute is invalid");
-    }
+
+  if (auto maximum_file_age = context->getProperty<core::TimePeriodValue>(MaximumFileAge)) {
+    maximum_file_age_ = maximum_file_age->getMilliseconds();
+  } else {
+    logger_->log_error("Maximum File Age attribute is missing or invalid");
```

Comment:
An exception could happen that couldn't happen before: `getProperty` throws on invalid value, and the conversion got pushed into `getProperty`. An exception thrown from `onSchedule` prevents the processor from starting, while the old behavior was to log an error and proceed with whatever default was set in the member. Please keep the old behavior if the property is invalid.

Reply:
This one calls the
```
template<typename T = std::string>
std::enable_if_t<std::is_default_constructible<T>::value, std::optional<T>>
getProperty(const Property& property) const {
  T value;
  try {
    if (!getProperty(property.getName(), value)) return std::nullopt;
  } catch (const utils::internal::ValueException&) {
    return std::nullopt;
  }
  return value;
}
```
which doesnt throw just returns a nullopt if the value is invalid

Reply:
Sorry, I assumed that all of them throw on validation errors.